consensus/istanbul/core/commit.go

```diff
@@ -78,6 +78,10 @@ func (c *core) handleCommit(msg *message, src istanbul.Validator) error {
 		return errInvalidMessage
 	}
 
+	if vrank != nil {
+		vrank.AddCommit(commit, src)
```

Comment:
How about moving `AddCommit` after checking the commit? (After `valSet.CheckInSubList`) It might be an invalid commit since we only check `View` in `isTargetCommit`.

Reply:
We need to receive "late commits" in vrank. "late commits" can be filtered in `checkMessage()` call, so it needs to be up here.

An unwanted scenario:
- as soon as 2f+1 commits are received, the state becomes committed
- soon after, `startNewRound()` is called to increase the current view
- then, `checkMessage()` filters "late commits" because they have the previous view